File: sleuth.py
#3.Quantify the TPM of each CDS in each transcriptome using kallisto and use these results as input to find differentially expressed genes between the two timepoints (2pi and 6dpi) using the R package sleuth.
# Write the following details for each significant transcript (FDR < 0.05) to your log file, include a header row, and tab-delimit each item: target_id test_stat pval qval

import os

def run_sleuth():
    sleuth = 'sleuth.R'
    os.system(sleuth)
    with open('miniProject_Aditi_Patel/sleuth_output.txt') as sleuth_output:
        with open("miniProject.log", "a") as miniProject_log:
            for line in sleuth_output:
                miniProject_log.write(line)

File: sleuth.R
#load sleuth package 
library(sleuth)
#load the dplyr package for data frame filtering 
library(dplyr)
#load data.table package to read in table data  
library(data.table)

#read in the table describing samples and kallisto output 
stab<-read.table("miniProject_Aditi_Patel/kallisto_table.txt",header=TRUE,stringsAsFactors=FALSE,sep='\t')

#initialize sleuth object
so<-sleuth_prep(stab)


#fit a model comparing two conditions 
so<-sleuth_fit(so,~condition,'full')

#fit the reduced model to compare in the likelihood ratio test
so<-sleuth_fit(so,~1,'reduced')

#perform the likelihood ratio test for differential expression between conditions (2dpi & 6dpi)
so<-sleuth_lrt(so,'reduced','full')

#extract the test results from the sleuth object 
sleuth_table<-sleuth_results (so,'reduced:full','lrt',show_all=FALSE)

#filter most significant results (FDR/qval <0.05) and sort by pval 
sleuth_significant<-dplyr::filter(sleuth_table,qval<=0.05)%>%dplyr::arrange(pval)
sleuth_output<-sleuth_significant %>% select(target_id,test_stat,pval,qval)

#write FDR <0.05 transcripts to file 
#sleuth_output<-na.omit(sleuth_output)
write.table(sleuth_output,file='miniProject_Aditi_Patel/sleuth_output.txt', quote=FALSE,row.names=FALSE)


